src/Sanitize.php

```diff
@@ -613,6 +636,39 @@ private function get_cache(string $image_url = ''): DataCache
 
         return $this->cache;
     }
+
+    /**
+     * Get a HTTP client
+     */
+    private function get_http_client(): Client
+    {
+        if ($this->http_client === null) {
+            $this->http_client = new FileClient(
```

Comment:
Looks like `SimplePie` will no longer share input data. Perhaps we should have separate internal `set_http_client` method that will continue accepting it from `SimplePie` and a public one `set_http_factories` for consumers that want to use `Sanitize` or `Locator` outside the context of `SimplePie`?

Reply:
`Sanitize` has already `set_http_client()` for PSR classes and the (deprecated) `pass_file_data()`. If one has not called `set_http_client()`, the http client will be created on the deprecated file data.